package/ios/Sources/RNCandle.swift

```diff
@@ -73,52 +78,134 @@ final class HybridRNCandle: HybridRNCandleSpec {
+              routingNumber: wireDetails.routingNumber
+            )
+          }
+          return AssetAccount(
+            legalAccountKind: legalAccountKind,
+            nickname: model.nickname,
+            details: .init(
+              fiatAccountDetails: .init(
+                assetKind: fiatDetails.assetKind.rawValue,
+                serviceAccountID: fiatDetails.serviceAccountID,
+                currencyCode: fiatDetails.currencyCode,
+                balance: fiatDetails.balance,
+                ach: ach,
+                wire: wire,
+                linkedAccountID: fiatDetails.linkedAccountID,
+                service: fiatDetails.service.toService),
+              marketAccountDetails: nil
+            )
+          )
+        case .MarketAccountDetails(let marketDetails):
+          return AssetAccount(
+            legalAccountKind: legalAccountKind,
+            nickname: model.nickname,
+            details: .init(
+              fiatAccountDetails: nil,
+              marketAccountDetails: .init(
+                assetKind: marketDetails.assetKind.rawValue,
+                serviceAccountID: marketDetails.serviceAccountID,
+                linkedAccountID: marketDetails.linkedAccountID,
+                service: marketDetails.service.toService
+              )
+            )
+          )
+        }
+      }
     }
   }
 
-  public func getTrades(span: String?) throws -> Promise<String> {
+  public func getTrades(query: TradeQuery) throws -> Promise<[Trade]> {
     .async {
-      return ""
+      let trades = try await self.viewModel.candleClient.getTrades(
+        query: .init(
+          linkedAccountIDs: query.linkedAccountIDs,
+          dateTimeSpan: query.dateTimeSpan,
+          gainedAssetKind: .init(rawValue: query.gainedAssetKind ?? ""),
+          lostAssetKind: .init(rawValue: query.lostAssetKind ?? ""),
+          counterpartyKind: .init(rawValue: query.counterpartyKind ?? "")
+        ))
+      return trades.map { trade in
+        return Trade(
+          dateTime: trade.dateTime,
+          state: TradeState(fromString: trade.state.rawValue)!,
+          counterparty: trade.toCounterparty,
+          lost: trade.lost.toAsset,
+          gained: trade.gained.toAsset
+        )
+      }
     }
   }
 
-  func getTradeQuotes(span: String?) throws -> Promise<String> {
+  public func getTradeQuotes(request: TradeQuoteRequest) throws -> Promise<[TradeQuote]> {
     .async {
-      return ""
+      let accounts = try await self.viewModel.candleClient.getTradeQuotes(
+        request:
+          .init(
+            linkedAccountIDs: request.linkedAccountIDs,
+            gained: try request.toGained
+          )
+      )
+      return accounts.map { account in
+        TradeQuote(
+          lost: account.lost.toAsset,
+          gained: account.gained.toAsset
+        )
+      }
     }
   }
 
+  public func submitTrade(serviceTradeID: String) throws -> Promise<TradeResult> {
+    fatalError()
```

Comment:
#req just `throw`, we shouldn't crash their app

Reply:
I didn't expose this function publicly, but sounds good.